### PR TITLE
Feature: Pure Go Lua (VM)

### LINKED ISSUE #288: Replace CGO Lua with GopherLua
The Lua support as it is, works, but relies on CGO, which makes the embedding of LedisDB no pure Go, the way it is supposed to be (possible).

Would it be possible to replace https://github.com/siddontang/golua with https://github.com/yuin/gopher-lua? I would be willing to put dev time into making this happen as a priority, as the dependency on CGO isn't desired.

Please let me know your opinion on this @siddontang and thank you very much for the time and effort you've put into ledisdb!

### PR DESCRIPTION
**Pull request by GlenDC**
This PR replaces the CGO Lua VM (https://github.com/siddontang/golua) with a Lua VM written in pure Golang (https://github.com/yuin/gopher-lua), and fixes #288 

Because of this the `lua` build tag is no longer needed, and neither is any of the CGO information related to the C Lua VM.

I also updated the documentation to reflect this PR, and the previous PR (#289).

I've split up this PR in 2 commits, the first one with the dependency changes (vendor) and the second one with the actual changes, this way you can more easily review this PR by simply focussing on the diff of the second commit.

### CODE REVIEW

**Comment by siddontang** [comment]
Cool @GlenDC 

Do you use `make update_vendor` to re-update the dependency? seem that some needless files are introduced in vendor.

**Comment by GlenDC** [comment]
@siddontang oh no sorry, totally missed that _Make_ target. Was manually updating the dependencies using the glide tool, and than copying them over to the `_vendor` folder. Will force update the PR, with the first commit replaced by a new commit.

**Comment by GlenDC** [comment]
@siddontang you were totally right, and thanks for providing that make target, I wish I had seen that earlier, hehe. The dependency commit is now last (instead for first), and contains indeed quite some files less, than the original first commit. Good for pointing that out!

**Comment by GlenDC** [comment]
@siddontang I also added support for CJSON, as that was also supported previously before. I think this PR is good to go now. Let me know what you think.

File: server/cmd_script.go [hunk 1]
```diff
@@ -71,39 +67,48 @@ func evalGenericCommand(c *client, evalSha1 bool) error {
...
+	l.Push(global)
 
-		if v, ok := r.(error); ok {
-			return v
+	// catch any uncaught panic
+	// this happens for example when the user,
+	// makes a mistake using `ledis.call`
+	defer func() {
+		if r := recover(); r != nil {
+			switch x := r.(type) {
```
Comment (siddontang) [inline server/cmd_script.go]: I think using `err = fmt.Errorf("panic: %v", r)` is enough.

**Comment by siddontang** [comment]
Thanks @GlenDC 

Seem that Go Lua is more easy to use. 

Rest LGTM

**Comment by GlenDC** [comment]
Thank you to @siddontang, for the fast responses, I really appreciate that.
I agree that using Gopher-Lua is quite easy, compared to the CGO lua.

I added and rebased the change you requested 👍

**Pull request merged**